test/gui/shared/scripts/pageObjects/SyncConnection.py

```diff
@@ -142,3 +151,20 @@ def confirm_folder_sync_connection_removal():
         squish.clickButton(
             squish.waitForObject(SyncConnection.REMOVE_FOLDER_SYNC_CONNECTION_BUTTON)
         )
+
+    @staticmethod
+    def wait_for_error_label(to_exist=True):
+        """Wait for permission error label to appear or disappear"""
+        status = squish.waitFor(
+            lambda: object.exists(SyncConnection.PERMISSION_ERROR_LABEL) == to_exist,
+            SyncConnection.PERMISSION_ERROR_WAIT_TIMEOUT * 1000
+        )
+        if not status:
+            action = "appear" if to_exist else "disappear"
+            raise AssertionError(f"Permission error label did not {action}")
+
+    @staticmethod
+    def get_permission_error_message():
+        """Get the permission error message text"""
+        SyncConnection.wait_for_error_label(True)  # Wait for label to appear
+        return str(squish.waitForObject(SyncConnection.PERMISSION_ERROR_LABEL, get_config("lowestSyncTimeout") * 1000).text)
```

Comment:
you can keep the default timeout
```suggestion
        return str(squish.waitForObject(SyncConnection.PERMISSION_ERROR_LABEL).text)
```